src/fsm/main.c

```diff
@@ -645,6 +658,7 @@ main(int argc, char *argv[])
 	}
 
 	fsm_free(fsm);
+	fsm_to_cleanup = NULL;
```

Comment:
Why not use the atexit proc instead?

Reply:
It's a good question.  I don't particularly like the `atexit()` code.  I added it to silence LSAN false positives and avoid adding a lot of cleanup code for all exit paths.  I'm not 100% sure that this should go into the actual code.

So... this is a bit of a style question for @katef and others (eg: @silentbicycle).

Reply:
Hiya, I'm fine with using atexit where we need to. In most cases I don't think we need to.